Documentation/project-docs/writing-tests.md

```diff
@@ -0,0 +1,93 @@
+This document provides an overview of paradigm we are using in our tests.
+
+# RemoteExecutor
```

Comment:
> don't have another way to isolate mentioned tests (static properties, whatever...).

I'm not sure the listed items are the reason for this application. Isolating statics can be achieved by serializing XUnit test execution. (https://xunit.github.io/docs/running-tests-in-parallel.html or using locking, etc)

I think the only reason this app exists is process-level isolation for reliable Diagnostics collection and shouldn't be used for other purposes.